python-sdk/src/astro/files/locations/google/gcs.py

```diff
@@ -35,7 +35,12 @@ def paths(self) -> list[str]:
 
     @property
     def size(self) -> int:
-        return -1
+        url = urlparse(self.path)
+        bucket_name = url.netloc
+        object_name = url.path
+        if object_name[0] == "/":
```

Comment:
What would happen when path is like `gs://bucket/*` ? We should add a test for this

Reply:
if you pass `gs://bucket/*` it would probably not work. 
because GCS hook `get_size` uses `get_blob` which expects blob name i.e object name and `*` is not the correct object name here
https://github.com/apache/airflow/blob/main/airflow/providers/google/cloud/hooks/gcs.py#L776
https://googleapis.dev/python/storage/latest/buckets.html

Reply:
infact even our paths property on location  will return an empty list